File: java/src/sdk/dataobjects/paypal/payflow/BrowserInfo.java
/*
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package paypal.payflow;

public final class BrowserInfo extends BaseRequestDataObject {

    private String browserTime;

    private String browserCountryCode;

    private String browserUserAgent;

    private String custom;

    private String buttonSource;

    private String notifyURL;

    private String merchantSessionId;

    /**
     * Used for Browser related information.
     * <p>Use the BrowserInfo object for the user
     * browser related information.</p>
     *
     * @paypal.sample <p>Following example shows how to use a
     * Browser Info object.</p>
     * .................
     * // inv is the Invoice object
     * .................
     * // Set the Browser Info details.
     * BrowserInfo browser = New BrowserInfo();
     * browser.setBrowserCountryCode ("USA");
     * browser.setBrowserUserAgent ("IE 6.0");
     * inv.setBrowserInfo (browser);
     * .................
     */
    public BrowserInfo() {
    }


    protected void generateRequest() {
        try {
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_BROWSERTIME, browserTime));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_BROWSERCOUNTRYCODE, browserCountryCode));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_BROWSERUSERAGENT, browserUserAgent));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_BUTTONSOURCE, buttonSource));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_CUSTOM, custom));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_NOTIFYURL, notifyURL));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MERCHANTSESSIONID, merchantSessionId));
        } catch (Exception ex) {
            ErrorObject error = PayflowUtility.populateCommError(PayflowConstants.E_UNKNOWN_STATE, ex, PayflowConstants.SEVERITY_FATAL, false, null);
            getContext().addError(error);
        }
    }

    /**
     * Gets the browser's country code.
     *
     * @return browserCountryCode String
     * @paypal.sample <p>Maps to Payflow Parameter: BROWSERCOUNTRYCODE</p>
     */
    public String getBrowserCountryCode() {
        return browserCountryCode;
    }

    /**
     * Sets the browser's country code.
     *
     * @param browserCountryCode String
     * @paypal.sample <p>Maps to Payflow Parameter: BROWSERCOUNTRYCODE</p>
     */
    public void setBrowserCountryCode(String browserCountryCode) {
        this.browserCountryCode = browserCountryCode;
    }

    /**
     * Gets the browser time.
     * <p>Browser's local time.</P>
     *
     * @return browserTime String
     * @paypal.sample <p>Maps to Payflow Parameter: BROWSERTIME</p>
     */
    public String getBrowserTime() {
        return browserTime;
    }

    /**
     * Sets the browser time.
     * <p>Browser's local time.</P>
     *
     * @param browserTime String
     * @paypal.sample <p>Maps to Payflow Parameter: BROWSERTIME</p>
     */
    public void setBrowserTime(String browserTime) {
        this.browserTime = browserTime;
    }

    /**
     * Gets the browser user agent.
     *
     * @return browserUserAgent String
     * @paypal.sample <p>Maps to Payflow Parameter: BROWSERUSERAGENT</p>
     */
    public String getBrowserUserAgent() {
        return browserUserAgent;
    }

    /**
     * Sets the browser user agent.
     *
     * @param browserUserAgent String
     * @paypal.sample <p>Maps to Payflow Parameter: BROWSERUSERAGENT</p>
     */
    public void setBrowserUserAgent(String browserUserAgent) {
        this.browserUserAgent = browserUserAgent;
    }

    /**
     * Gets the ButtonSource for Direct Payment and Express checkout.
     *
     * @return buttonSource String
     *         <p>Maps to Payflow Parameter : BUTTONSOURCE</p>
     */
    public String getButtonSource() {
        return buttonSource;
    }

    /**
     * Sets the ButtonSource for Direct Payment and Express Checkout.
     *
     * @param buttonSource String
     *                     <p>Maps to Payflow Parameter : BUTTONSOURCE</p>
     */
    public void setButtonSource(String buttonSource) {
        this.buttonSource = buttonSource;
    }

    /**
     * Gets Custom for Direct Payment and Express Checkout.
     *
     * @return custom String
     *         <p>Maps to Payflow Parameter : CUSTOM</p>
     */
    public String getCustom() {
        return custom;
    }

    /**
     * Sets Custom for Direct Payment and Express Checkout.
     *
     * @param custom String
     *               <p>Maps to Payflow Parameter : CUSTOM</p>
     */
    public void setCustom(String custom) {
        this.custom = custom;
    }

    /**
     * Gets the merchantSessionId for Direct Payment and Express Checkout.
     *
     * @return merchantSessionId String
     *         <p>Maps to Payflow Parameter : MERCHANTSESSIONID</p>
     */
    public String getMerchantSessionId() {
        return merchantSessionId;
    }

    /**
     * Sets the merchantSessionId for Direct Payment and Express Checkout.
     *
     * @param merchantSessionId String
     *                          <p>Maps to Payflow Parameter : MERCHANTSESSIONID</p>
     */
    public void setMerchantSessionId(String merchantSessionId) {
        this.merchantSessionId = merchantSessionId;
    }

    /**
     * Gets the notifyURL String for Direct Payments and Express Checkout.
     *
     * @return notifyURL String
     *         <p>Maps to Payflow Parameter : NOTIFYURL</p>
     */
    public String getNotifyURL() {
        return notifyURL;
    }

    /**
     * Sets the notifyURL for DirectP ayments and Express Checkout.
     *
     * @param notifyURL String
     *                  <p>Maps to Payflow Parameter : NOTIFYURL</p>
     */
    public void setNotifyURL(String notifyURL) {
        this.notifyURL = notifyURL;
    }

}

File: dotNET/PayflowProSDK/PayPal/Payments/DataObjects/BrowserInfo.cs
#region "Imports"

using System;
using PayPal.Payments.Common.Utility;
using PayPal.Payments.Common.Exceptions;

#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// Used for Browser related information.
	/// </summary>
	/// <remarks>Use the BrowserInfo object for the user
	/// browser related information.</remarks>
	/// <example>
	/// <para>Following example shows how to use a 
	/// Browser Info object.</para>
	/// <code lang="C#" escaped="false">
	///	.................
	///	// Inv is the Invoice object
	///	.................
	///	// Set the Browser Info details.
	///	BrowserInfo Browser = New BrowserInfo();
	///	Browser.BrowserCountryCode = "USA";
	///	Browser.BrowserUserAgent = "IE 6.0";
	///	Inv.BrowserInfo = Browser;
	///	.................
	///	</code>
	/// <code lang="Visual Basic" escaped="false">
	///	.................
	///	' Inv is the Invoice object
	///	.................
	///	' Set the Browser Info details.
	///	Dim Browser As BrowserInfo = New BrowserInfo
	///	Browser.BrowserCountryCode  = "USA"
	///	Browser.BrowserUserAgent = "IE 6.0"
	///	Inv.BrowserInfo = Browser
	///	.................
	///	</code>
	/// </example>
	public sealed class BrowserInfo : BaseRequestDataObject
	{
		#region "Member Variables"

		/// <summary>
		/// Holds Browser time.
		/// </summary>
		private String mBrowserTime;
		/// <summary>
		/// Holds Browser Country code.
		/// </summary>
		private String mBrowserCountryCode;
		/// <summary>
		/// Holds Browser User Agent
		/// </summary>
		private String mBrowserUserAgent;
		/// <summary>
		/// Holds Buttonsource
		/// </summary>
		private String mButtonSource;
		/// <summary>
		/// Holds notify url
		/// </summary>
		private String mNotifyURL;
		/// <summary>
		/// Holds custom
		/// </summary>
		private String mCustom;
		/// <summary>
		/// Holds merchant session id
		/// </summary>
		private String mMerchantSessionId;
		#endregion

		#region "Constructors"

		/// <summary>
		/// Constructor for BrowserInfo
		/// </summary>
		/// <remarks>Use the BrowserInfo object for the user
		/// browser related information.</remarks>
		/// <example>
		/// <para>Following example shows how to use a 
		/// Browser Info object.</para>
		/// <code lang="C#" escaped="false">
		///	.................
		///	// Inv is the Invoice object
		///	.................
		///	// Set the Browser Info details.
		///	BrowserInfo Browser = New BrowserInfo();
		///	Browser.BrowserCountryCode = "USA";
		///	Browser.BrowserUserAgent = "IE 6.0";
		///	Inv.BrowserInfo = Browser;
		///	.................
		///	</code>
		/// <code lang="Visual Basic" escaped="false">
		///	.................
		///	' Inv is the Invoice object
		///	.................
		///	' Set the Browser Info details.
		///	Dim Browser As BrowserInfo = New BrowserInfo
		///	Browser.BrowserCountryCode  = "USA"
		///	Browser.BrowserUserAgent = "IE 6.0"
		///	Inv.BrowserInfo = Browser
		///	.................
		///	</code>
		/// </example>
		public BrowserInfo()
		{
		}

		#endregion

		#region "Properties"

		/// <summary>
		/// Gets, Sets Browser time.
		/// </summary>
		/// <remarks>
		/// <para>Browser's local time.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BROWSERTIME</code>
		/// </remarks>
		public String BrowserTime
		{
			get { return mBrowserTime; }
			set { mBrowserTime = value; }
		}

		/// <summary>
		/// Gets, Sets Browser Country code.
		/// </summary>
		/// <remarks>
		/// <para>Browser's local Country Code.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BROWSERCOUNTRYCODE</code>
		/// </remarks>
		public String BrowserCountryCode
		{
			get { return mBrowserCountryCode; }
			set { mBrowserCountryCode = value; }
		}

		/// <summary>
		/// Gets, Sets Browser user agent.
		/// </summary>
		/// <remarks>
		/// <para>Browser's user agent.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BROWSERUSERAGENT</code>
		/// </remarks>
		public String BrowserUserAgent
		{
			get { return mBrowserUserAgent; }
			set { mBrowserUserAgent = value; }
		}

		/// <summary>
		/// Gets or Sets the custom parameter for Direct Payment and Express checkout.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>CUSTOM</code>
		/// </remarks>
		public String Custom
		{
			get{return mCustom; }
			set{mCustom= value;}
		}

		/// <summary>
		/// Gets or Sets the buttonsource parameter for Direct Payment and Express checkout.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BUTTONSOURCE</code>
		/// </remarks>
		public String ButtonSource
		{
			get{return mButtonSource; }
			set{mButtonSource = value;}
		}

		/// <summary>
		/// Gets or Sets the NotifyUrl parameter for Direct Payment and Express checkout.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>NOTIFYURL</code>
		/// </remarks>
		public String NotifyURL
		{
			get{return mNotifyURL; }
			set{mNotifyURL = value;}
		}
		/// <summary>
		/// Gets or Sets the MerchantSessionId parameter for Direct Payment.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>MERCHANTSESSIONID</code>
		/// </remarks>
		public String MerchantSessionId
		{
			get{return mMerchantSessionId; }
			set{mMerchantSessionId = value;}
		}

		#endregion

		#region "Core functions"

		/// <summary>
		/// Generates the transaction request.
		/// </summary>
		internal override void GenerateRequest()
		{
			try
			{
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_BROWSERTIME, mBrowserTime));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_BROWSERCOUNTRYCODE, mBrowserCountryCode));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_BROWSERUSERAGENT, mBrowserUserAgent));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_CUSTOM, mCustom));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_BUTTONSOURCE, mButtonSource));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_NOTIFYURL, mNotifyURL));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MERCHANTSESSIONID, mMerchantSessionId));
			}
			catch (BaseException)
			{
				throw;
			}
			catch (Exception Ex)
			{
				DataObjectException DEx = new DataObjectException(Ex);
				throw DEx;
			}
            //catch
            //{
            //    throw new Exception();				
            //}
            }
        		#endregion
	}
}